provider/image.go

```diff
@@ -585,20 +594,25 @@ func marshalBuildAndApplyDefaults(b resource.PropertyValue) (Build, error) {
 	buildObject := b.ObjectValue()
 
 	// Context
-	if buildObject["context"].IsNull() {
-		// set default
-		build.Context = "."
-	} else {
-		build.Context = buildObject["context"].StringValue()
+	if !buildObject["context"].ContainsUnknowns() {
+		if buildObject["context"].IsNull() {
+			// set default
+			build.Context = "."
+		} else {
+			build.Context = buildObject["context"].StringValue()
+		}
 	}
 
 	// Dockerfile
-	if buildObject["dockerfile"].IsNull() {
-		// set default
-		build.Dockerfile = path.Join(build.Context, defaultDockerfile)
-	} else {
-		build.Dockerfile = buildObject["dockerfile"].StringValue()
+	if !buildObject["dockerfile"].ContainsUnknowns() {
+		if buildObject["dockerfile"].IsNull() {
+			// set default
+			build.Dockerfile = path.Join(build.Context, defaultDockerfile)
+		} else {
+			build.Dockerfile = buildObject["dockerfile"].StringValue()
+		}
 	}
+
```

Comment:
So if I understand correctly, we'll only run a preview as long as all of our inputs have resolved. That seems reasonable. I think that matches how the bridge provider previously did these previews?